source/portable/mbedtls/core_pkcs11_mbedtls.c

```diff
@@ -4849,13 +4726,16 @@ CK_DECLARE_FUNCTION( CK_RV, C_GenerateKeyPair )( CK_SESSION_HANDLE hSession,
 
             if( xResult != CKR_OK )
             {
-                ( void ) PKCS11_PAL_DestroyObject( *phPrivateKey );
-                LogDebug( ( "Destroyed %lu private key handle due to errors.", ( unsigned long int ) *phPrivateKey ) );
+                ( void ) PKCS11_PAL_DestroyObject( xPalPrivate );
```

Comment:
Why are we throwing away the return values here?  Same comment at line 4730/4736/4738.  Shouldn't we log an error message if this is failing?  Discarding these and failing silently makes it harder to diagnose any issues which might occur.

Reply:
Fixed 👍